pkg/model/crd.go

```diff
@@ -757,6 +757,17 @@ func (r *CRD) HasMember(
 	return inSpec, inStatus
 }
 
+// HasReferenceFields returns true if any of the fields in CRD is a reference
+// field. Otherwise returns false
+func (r *CRD) HasReferenceFields() bool {
+	for _, fields := range r.Fields {
+		if fields.IsReference() {
```

Comment:
No, :) . 
Since this is a method on CRD, i.e. whether CRD "HasReferenceFields", so we iterate over each field to see if there "IsReference" field or not.